enterprise/ha/src/main/java/org/neo4j/kernel/ha/HaSettings.java

```diff
@@ -47,9 +47,14 @@
     @Description( "How long a slave will wait for response from master before giving up." )
     public static final Setting<Long> read_timeout = setting( "ha.read_timeout", DURATION, "20s" );
 
-    @Description( "Timeout for waiting for instance to become master or slave." )
+    @Description( "Timeout for request threads waiting for instance to become master or slave." )
     public static final Setting<Long> state_switch_timeout = setting( "ha.state_switch_timeout", DURATION, "120s" );
 
+    @Description( "Timeout for waiting for internal conditions during state switch, like for transactions "
+            + "to complete, before switching to master or slave." )
+    public static final Setting<Long> internal_state_switch_timeout =
+    setting( "ha.internal_state_switch_timeout", DURATION, "10s" );
```

Comment:
Somehow the indenting was lost.

This timeout is unrelated to heartbeats though. This is the max time a slave will wait between getting elected and starting the switch to master, waiting for active transaction to complete. The longer this timeout, the longer we've got a master-less cluster.